### ISSUE TITLE
Support Request: Need help/examples for the mqtt part

### ISSUE THREAD

**Issue by HolgerAusB** [OpenWonderLabs/homebridge-switchbot#685]
### Describe Your Problem

I don't know what i can put in the two option fields, and how

@banboobee as far as I understand, you wrote the mqtt part. Please could you provide a little guide or help page in the wiki?

### MQTT Url
 The field hint just shows an IP but you need to give a full URL with `mqtt://192.168.x.x`. Does it work when using tls by `mqtts:` instead?

### MQTT Options
What are the possible options and how to place more than one? I think one can put username, password, psk and other mqtt-stuff, or?

### MQTT Pub Options
Are there any other options other than retain?

### Device & Model

meter (EU)

### Homebridge SwitchBot Plugin Version

2.5.2

### Operating System

Raspbian Bullseye

**Comment by banboobee** [comment]
OPTIONS can be specified in a JSON form of {"a": x, "b"; y}, and ,as described in the PR, following repository details the OPTIONS, 

https://github.com/mqttjs/MQTT.js

More preciously, for mqttOptions,

https://github.com/mqttjs/MQTT.js#mqttclientstreambuilder-options

and for mqttPubOptions,

https://github.com/mqttjs/MQTT.js#mqttclientpublishtopic-message-options-callback

also for URL scheme,

https://github.com/mqttjs/MQTT.js#mqttconnecturl-options

**Comment by banboobee** [comment]
OPTIONS can be specified in a JSON form of {"a": x, "b": y}, and ,as described in the PR, following repository details the OPTIONS,

https://github.com/mqttjs/MQTT.js

More precisely, for mqttOptions,

https://github.com/mqttjs/MQTT.js#mqttclientstreambuilder-options

and for mqttPubOptions,

https://github.com/mqttjs/MQTT.js#mqttclientpublishtopic-message-options-callback

also for URL scheme,

https://github.com/mqttjs/MQTT.js#mqttconnecturl-options

**Comment by HolgerAusB** [comment]
Tanks for your help @banboobee. With these informations I could manage all I need.

But that does not explain how I would put this options into the input boxes of the plugin's UI. I need to put that directly to the JSON config.

So I ask you again, if you want to make a little wiki article for that in your PR or the issues.

BTW: is there any chance we can get a battery level exposed as well?

**Comment by HolgerAusB** [comment]
Sorry @banboobee but I think there is a bug. In the UI, there are three fields for MQTT URL, MQTT Options and MQTT Pub Options.

But both of the option fields do not work. If I try to put a JSON-object to that fields all `"` are escaped by `\"`. It worked, when I put the obejects directly to the plugins JSON config. But if I now change anything else in the plugin config via UI, all mqtt option objects are stripped from the config.

Now I ended up with an error code in the log, even if I remove all mqtt things. I think I had to re-setup the complete switchbot plugin now. It's just 4 object, so no problem on my site, but others maybe won't like to do.

Config example:
```
{
  "deviceId": "123456789012",
  "configDeviceName": "MeterPlus",
  "configDeviceType": "MeterPlus",
  "mqttURL": "mqtts://mqtt.example.com",
  "mqttOptions": {
      "port": 8883,
      "username": "hb-switchbot",
      "password": "very-secret"
  },
  "mqttPubOptions": {
      "retain": true
  },
  "refreshRate": 30
},
```

The Error in Log:
`[SwitchBot] API Request: TypeError: Cannot read properties of undefined (reading 'map'), Submit Bugs Here: https://tinyurl.com/SwitchBotBug`
